### PR TITLE
Manual edits by MGB

### PR DESCRIPTION
**Pull request by mgbellemare**
I edited bits of the manual, mostly for style and conciseness. Also revamped the RL-Glue and fifo interface sections. Please have a quick look.

### CODE REVIEW

File: doc/manual/manual.tex [hunk 1]
```diff
@@ -77,81 +70,56 @@ \subsubsection*{Linux (e.g.: Ubuntu):}
...
-the actions that do have some effect on that game. To obtain such list of actions one has to call, for
-example:\\
+There are two different action sets provided by ALE: the ``legal'' set and the ``minimal'' 
+set. Save for a few rare exceptions, the legal action set consists of all 18 actions for all games, including duplicates and actions with no effect. On the other hand, the minimal action set for a game contains only 
+the actions that have some effect on that game. The \verb+getLegalActionSet+ and \verb+getMinimalActionSet+ methods provide the desired action sets:\\ 
 
 \verb+ActionVect legal_actions = ale.getLegalActionSet();+\\
 
-Then, to act one can call the function \verb+act()+ passing an object of \verb+Action+ as parameter:\\
+Taking ana ction is done by calling the function \verb+act()+ passing an object of \verb+Action+ as a parameter:\\
```
Comment (mcmachado) [inline doc/manual/manual.tex]: Typo: Taking an action

File: doc/manual/manual.tex [hunk 2]
```diff
@@ -20,56 +20,49 @@
...
-Notice that in the past (previous to version 0.5.0) the recommended method to compile the ALE was
-through Makefiles written by the ALE developers. This has changed and now we suggest everyone to
-use CMake to generate a Makefile and then to compile the code using such generated Makefile. The
-old Makefiles written for Linux and OS X (files \verb+makefile.mac+ and \verb+makefile.unix+) are
-still available, but we do not support them anymore.
-
-The ALE provides optional functionalities that are not activated by default, but if chosen the user is 
-required to use additional packages:
+
+Two custom makefiles are provided, \verb+makefile.mac+ (Mac OS X) and \verb+makefile.unix+ (Linux). However, beginning with version 0.5, we highly recommend the use of CMake rather than the custom makefiles.
```
Comment (mcmachado) [inline doc/manual/manual.tex]: Here is the unique place you refer to the new version as 0.5. All the other places you refer to it as 0.5.0.

File: doc/manual/manual.tex [hunk 3]
```diff
@@ -500,50 +460,190 @@ \subsection{Minimal Action Set}
...
+
+ffmpeg version 2.6.1 Copyright (c) 2000-2015 the FFmpeg developers
+  built with Apple clang version 4.1 (tags/Apple/clang-421.11.65) ... 
+
+[ loads of output ]
+
+ale_0_5> open agent.mov
+\end{verbatim}
+
+\section{Troubleshooting / FAQ} \label{sec:troubleshooting}
```
Comment (mcmachado) [inline doc/manual/manual.tex]: Is it worth to mention here the mailing list, so people can ask additional questions?

**Comment by mgbellemare** [comment]
Edited Marlos' comments directly into this pull request.

**Pull request merged**